src/components/Search/index.tsx

```diff
@@ -57,44 +51,39 @@ const handleSearchResult = (
   }
 
   setSearchLoading(inputElement, t)
-  explorerService.api
-    .fetchSearchResult(addPrefixForHash(query))
-    .then((response: any) => {
-      const { data } = response
-      if (!response || !data.type) {
-        history.push(`/search/fail?q=${query}`)
-        return
-      }
+
+  try {
+    const { data } = await explorerService.api.fetchSearchResult(addPrefixForHash(query))
+    clearSearchInput(inputElement)
+    setSearchValue('')
+    if (data.type === SearchResultType.Block) {
+      history.push(`/block/${data.attributes.blockHash}`)
+    } else if (data.type === SearchResultType.Transaction) {
+      history.push(`/transaction/${data.attributes.transactionHash}`)
+    } else if (data.type === SearchResultType.Address) {
+      history.push(`/address/${data.attributes.addressHash}`)
+    } else if (data.type === SearchResultType.LockHash) {
+      history.push(`/address/${data.attributes.lockHash}`)
+    } else if (data.type === SearchResultType.UDT) {
+      history.push(`/sudt/${query}`)
+    }
+  } catch (error) {
+    setSearchContent(inputElement, query)
+
+    if (
+      isAxiosError(error) &&
+      error.response?.data &&
```

Comment:
```suggestion
      error.response &&
      error.response.data &&
```

Reply:
The type of response here is still Nullable, probably because AxiosError might be thrown before and during the request?

```typescript
export interface AxiosError<T = any> extends Error {
  ...,
  response?: AxiosResponse<T>;
  ...,
}
```